### PR TITLE
Remove POC Form primitive

### PR DESCRIPTION
**Pull request by reesscot**
*Description of changes:*
Removes the `Form` primitive and replaces it with `form` in all the places it was used in Authenticator.

TODO:

- [x] Test locally in next-example apps for style regressions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/react/src/helpers/utils.ts [hunk 1]
```diff
@@ -1 +1,5 @@
 export const isDevelopment = () => process.env.NODE_ENV !== 'production';
+
+export const isInputTarget = (target: unknown): target is HTMLInputElement => {
```
Comment (hvergara) [inline packages/react/src/helpers/utils.ts:3]: you are going to hate me, but: can we rename it to `isInputEventTarget`? current name seems too generic IMO

**Review by hvergara — CHANGES_REQUESTED** [review]

File: packages/react/src/helpers/utils.ts (hunk 1 above)
Comment (reesscot) [reply packages/react/src/helpers/utils.ts]: No problem, updated

File: packages/react/src/helpers/utils.ts [hunk 2]
```diff
@@ -1 +1,7 @@
 export const isDevelopment = () => process.env.NODE_ENV !== 'production';
+
+export const isInputEventTarget = (
+  target: unknown
+): target is HTMLInputElement => {
+  return (target as HTMLInputElement)?.nodeName === 'INPUT';
+};
```
Comment (ericclemmons) [inline packages/react/src/helpers/utils.ts:7]: Was there a bug or type issue?  I'm curious what you saw that we may have missed so far...

**Review by ericclemmons — APPROVED** [review]
Thanks for doing this!

Nothing blocking, but curious what prompted the creation of `isInputEventTarget`

File: packages/react/src/helpers/utils.ts (hunk 2 above)
Comment (reesscot) [reply packages/react/src/helpers/utils.ts:7]: Yes, changing from `Form` to `form` exposed that the `handleChange` function had the wrong type on its `event` parameter, which caused type errors on `form.onChange` and when trying to de-structure the `event.target`. The form onchange event doesn't know what type of element fired the change event, so the `isInputEventTarget` typeguard ensures it was an input element.

```diff
-  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+ const handleChange = (event: React.FormEvent<HTMLFormElement>) => {
```

File: packages/react/src/helpers/utils.ts [hunk 3]
```diff
@@ -1 +1,14 @@
...
+): target is HTMLInputElement | HTMLSelectElement => {
+  return (
+    (target as HTMLInputElement)?.nodeName === 'INPUT' ||
+    (target as HTMLSelectElement)?.nodeName === 'SELECT'
+  );
+};
+
+export const isInputElement = (target: unknown): target is HTMLInputElement => {
+  return (target as HTMLInputElement)?.nodeName === 'INPUT';
+};
```
Comment (slaymance) [inline packages/react/src/helpers/utils.ts:14]: Is it worthwhile to break out both input and select checks into separate utils? Could we use them in the future?

```typescript
export const isInputElement = (target: unknown): target is HTMLInputElement =>
  (target as HTMLInputElement)?.nodeName === 'INPUT';

export const isSelectElement = (target: unknown): target is HTMLSelectElement =>
  (target as HTMLSelectElement)?.nodeName === 'SELECT';

export const isInputOrSelectElement = overSome([isInputElement, isSelectElement]);
```

**Review by slaymance — APPROVED** [review]
Looks good to me!

File: packages/react/src/helpers/utils.ts (hunk 3 above)
Comment (reesscot) [reply packages/react/src/helpers/utils.ts:17]: Split them into separate functions

**Review by hvergara — APPROVED** [review]

**Pull request merged**